Assets/Scripts/Utility/UIManager.cs

```diff
@@ -0,0 +1,109 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+
+public class UIManager : MonoBehaviour
+{
+    // Var Player and Abilities
+    private GameObject player;
+    private GameObject abilities;
+    // Var HP
+    private GameObject playerHPBar;
+    private Image hpFilling;
+    private TMP_Text hpText;
+    // Var Mana
+    private GameObject playerManaBar;
+    private Image manaFilling;
+    private TMP_Text manaText;
+    // Var Teleport
+    private GameObject teleportObject;
+    private Image teleportShadow;
+    private float teleportTimer;
+    private float teleportCooldown;
+    // Var PierceAttack
+    private GameObject pierceAttackObject;
+    private Image pierceAttackShadow;
+    private float pierceAttackTimer;
+    private float pierceAttackCooldown;
+    // Var BasicAttack
+    private GameObject basicAttackObject;
+    private Image basicAttackShadow;
+    private float basicAttackTimer;
+    private float basicAttackCooldown;
+
+
+    void Start()
+    {
+        //Access to HP Bar
+        playerHPBar = transform.Find("HPBar").gameObject;
```

Comment:
the only time playerHPBar is used is to get the child and to check if null. This creates waste and adds more code than needed. Notice that theres a hardcoded value "HPBar" which is also bad. 

Use [SerializeField] to specifically get the hpFilling and hpText and assign them in the inspector.